src/sinol_make/helpers/compile.py

```diff
@@ -1,15 +1,57 @@
-from typing import Tuple, Union
+from typing import Tuple, Union, Dict
 import os
 import sys
 import shutil
 import stat
 import subprocess
 
 import sinol_make.helpers.compiler as compiler
+from sinol_make import util
 from sinol_make.interfaces.Errors import CompilationError
 from sinol_make.structs.compiler_structs import Compilers
 
 
+def get_executable_info_file(file_path):
+    """
+    Get the path to the file containing path to the compiled executable based on the file's md5 sum.
+    """
+    os.makedirs(os.path.join(os.getcwd(), 'cache', 'md5sums'), exist_ok=True)
+    md5sum = util.get_file_md5(file_path)
+    return os.path.join(os.getcwd(), 'cache', 'md5sums', md5sum)
+
+
+def check_compiled(file_path: str):
+    """
+    Check if a file is compiled
+    :param file_path: Path to the file
+    :return: executable path if compiled, None otherwise
+    """
+    info_file_path = get_executable_info_file(file_path)
+
+    try:
+        with open(info_file_path, 'r') as md5sums_file:
+            exe_file = md5sums_file.read().strip()
+            if os.path.exists(exe_file):
+                return exe_file
+            else:
+                os.unlink(info_file_path)
+                return None
+    except FileNotFoundError:
+        return None
+
+
+def save_compiled(file_path: str, exe_path: str):
+    """
+    Save md5 sum of file to cache, so we know it is compiled.
```

Comment:
```suggestion
    Save the md5sum of the path of the file being compiled, so we know it is compiled.
```
Let's make it less ambiguous (though I'm not sure whether I got it correctly).